### PR TITLE
feat(lazyloaidng): Added lazy loading and reponsive navbar

### PR DESCRIPTION
**Pull request by nirojneupane10**

### CODE REVIEW

File: frontend/src/components/Loader/index.tsx [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+const Loading: React.FC<LoadingProps> = ({ size, message }) => {
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        flexDirection: "column",
+        height: "100vh",
+      }}
```
Comment (adars-kafle) [inline frontend/src/components/Loader/index.tsx:18]: avoid inline styles, rather use styled components of MUI.

File: frontend/src/components/navbar/index.tsx [hunk 2]
```diff
@@ -15,7 +27,39 @@ const navbarItem: NavbarLinkProps[] = [
     path: "/product",
   },
 ];
+
```
Comment (adars-kafle) [inline frontend/src/components/navbar/index.tsx:30]: keep this in `constants/navs.ts`

File: frontend/src/components/navbar/index.tsx [hunk 3]
```diff
@@ -33,11 +77,20 @@ const Navbar: React.FC = () => {
...
+              </IconButton>
+              {drawer}
+            </>
+          ) : (
+            <Box sx={{ gap: 2 }} display="flex">
+              {navbarItem.map((item) => (
+                <NavbarLink key={item.name} name={item.name} path={item.path} />
+              ))}
+            </Box>
+          )}
```
Comment (adars-kafle) [inline frontend/src/components/navbar/index.tsx:93]: this file is quite big, decompose it!

**Review by adars-kafle — CHANGES_REQUESTED** [review]

**Comment by adars-kafle** [comment]
@nirojneupane10  you have merge conflicts, resolve them!!

File: frontend/src/components/Loader/index.tsx [hunk 4]
```diff
@@ -1,30 +1,34 @@
 import React from "react";
-import { Box, CircularProgress, Typography } from "@mui/material";
+import { Box, CircularProgress, Typography, styled } from "@mui/material";
 
 type LoadingProps = {
   size?: number;
   message?: string;
 };
```
Comment (adars-kafle) [inline frontend/src/components/Loader/index.tsx:8]: Please try to separate the types in `types/` directory!!

File: frontend/src/components/Loader/index.tsx [hunk 5]
```diff
@@ -1,30 +1,34 @@
...
+  justifyContent: "center",
+  alignItems: "center",
+  flexDirection: "column",
+  height: "100vh",
+}));
+
+const StyledTypography = styled(Typography)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+}));
+
```
Comment (adars-kafle) [inline frontend/src/components/Loader/index.tsx:20]: we can add these styled box and typography inside the `styles.ts` and import from there

File: frontend/src/components/navbar/Sidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import MenuIcon from "@mui/icons-material/Menu";
+
+const Sidebar = () => {
+  const [drawerOpen, setDrawerOpen] = useState(false);
+
+  const toggleDrawer = (open: boolean) => () => {
+    setDrawerOpen(open);
+  };
+
+  const drawer = (
```
Comment (adars-kafle) [inline frontend/src/components/navbar/Sidebar.tsx:21]: You still haven't changed the drawer naming. This is not the recommended name for jsx components.

**Review by adars-kafle — CHANGES_REQUESTED** [review]

File: frontend/src/components/LazyLoads.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React, { Suspense } from "react";
+import Loading from "./Loader";
+import ErrorBoundary from "./ErrorBoundary";
+
+interface LazyLoadProps {
+  Component: React.LazyExoticComponent<React.FC>;
+}
+
```
[the remark this answers is not in the document]
Comment (nirojneupane10) [reply frontend/src/components/LazyLoads.tsx:5]: provide the example

[the remark this answers is not in the document]
Comment (adars-kafle) [reply frontend/src/components/LazyLoads.tsx:5]: Create a file named `lazyLoading.ts` inside `types/` directory and:

```
export interface LazyLoadProps {
     Component: React.LazyExoticComponent<React.FC>;
}
```

and then you can import it in `LazyLoads.tsx` like:
```
import { LazyLoadProps } from "types/lazyLoad";
```

File: frontend/src/components/navbar/index.tsx [hunk 8]
```diff
@@ -15,7 +27,39 @@ const navbarItem: NavbarLinkProps[] = [
...
 const Navbar: React.FC = () => {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down("md"));
+  const [drawerOpen, setDrawerOpen] = useState(false);
+
+  const toggleDrawer = (open: boolean) => () => {
+    setDrawerOpen(open);
+  };
+
+  const drawer = (
```
[the remark this answers is not in the document]
Comment (nirojneupane10) [reply frontend/src/components/navbar/index.tsx]: Provide the example

File: frontend/src/components/navbar/Sidebar.tsx (hunk 6 above)
Comment (nirojneupane10) [reply frontend/src/components/navbar/Sidebar.tsx:21]: Example

File: frontend/src/components/Loader/index.tsx (hunk 5 above)
Comment (nirojneupane10) [reply frontend/src/components/Loader/index.tsx:20]: Provide the example when you are reviewing

File: frontend/src/components/Loader/index.tsx (hunk 1 above)
Comment (nirojneupane10) [reply frontend/src/components/Loader/index.tsx]: Provide the example

File: frontend/src/components/navbar/Sidebar.tsx (hunk 6 above)
Comment (adars-kafle) [reply frontend/src/components/navbar/Sidebar.tsx:21]: you should be familiar with it i guess. anyways here is the link for you to read:

https://www.linkedin.com/pulse/react-js-naming-convention-kristiyan-velkov/

File: frontend/src/components/Loader/index.tsx (hunk 5 above)
Comment (adars-kafle) [reply frontend/src/components/Loader/index.tsx:20]: It's simple import-export thing! I don't think any examples are required as i have clearly defined what to do above!!!

**Comment by adars-kafle** [comment]
@nirojneupane10 I see that your commit messages are in past tense which is not according to conventional commit standards. Read the varicon project documentation to learn more about commit messages and branching names!

File: frontend/src/routes/index.tsx [hunk 9]
```diff
@@ -5,18 +5,21 @@ import {
...
 
 const Routes = () => {
   const router = createBrowserRouter(
     createRoutesFromElements(
       <Route path="/" element={<Navbar />}>
-        <Route index element={<Home />} />
-        <Route path="/product" element={<Products />} />
+        <Route index element={<LazyLoad Component={Home} />} />
+        <Route path="/product" element={<LazyLoad Component={Products} />} />
       </Route>
```
Comment (adars-kafle) [inline frontend/src/routes/index.tsx:19]: Follow DRY principle!

**Pull request closed without merging**